### PR TITLE
feat: custom keybindings support

### PR DESCRIPTION
**Pull request by TiagoRibeiro25**
Added functionality to allow users to define custom keybindings.

This update introduces a new "keybindings" section in the configuration file, which contains the default keybindings for the application. Users can now modify these keybindings to suit their preferences. This enhancement addresses the issue reported in [#135](https://github.com/savedra1/clipse/issues/135).

### CODE REVIEW

File: config/constants.go [hunk 1]
```diff
@@ -16,11 +16,33 @@ const (
...
+			"choose":        "enter",
+			"remove":        "x",
+			"togglePin":     "p",
+			"togglePinned":  "tab",
+			"preview":       " ",
+			"selectDown":    "ctrl+down",
+			"selectUp":      "ctrl+up",
+			"selectSingle":  "s",
+			"clearSelected": "S",
+			"fuzzySelect":   "F",
```
Comment (savedra1) [inline config/constants.go:38]: fuzzySelect - again not sure what this is. maybe I'm forgetting my own code :laughing:

File: config/constants.go [hunk 2]
```diff
@@ -16,11 +16,33 @@ const (
...
 	return Config{
-		HistoryFilePath: defaultHistoryFile,
-		MaxHistory:      defaultMaxHist,
-		AllowDuplicates: defaultAllowDuplicates,
-		TempDirPath:     defaultTempDir,
-		LogFilePath:     defaultLogFile,
-		ThemeFilePath:   defaultThemeFile,
+		AllowDuplicates: true,
+		HistoryFilePath: "clipboard_history.json",
+		MaxHistory:      1000,
```
Comment (savedra1) [inline config/constants.go:21]: I think default limit should be kept to 100, as having a large history can cause performance issues for some machines.

File: config/constants.go [hunk 3]
```diff
@@ -16,11 +16,33 @@ const (
 // Because Go does not support constant Structs :(
 func defaultConfig() Config {
 	return Config{
-		HistoryFilePath: defaultHistoryFile,
-		MaxHistory:      defaultMaxHist,
-		AllowDuplicates: defaultAllowDuplicates,
-		TempDirPath:     defaultTempDir,
-		LogFilePath:     defaultLogFile,
-		ThemeFilePath:   defaultThemeFile,
+		AllowDuplicates: true,
```
Comment (savedra1) [inline config/constants.go:19]: I think this should default to false.

**Review by savedra1 — CHANGES_REQUESTED** [review]
Hi @TiagoRibeiro25 thank you so much for this! It's super appreciated :blue_heart: I have tested this locally and it works well. My notes are only for minor things, let me know what you think :smile:

File: config/constants.go [hunk 4]
```diff
@@ -16,11 +16,33 @@ const (
...
+		HistoryFilePath: "clipboard_history.json",
```
Comment (savedra1) [inline config/constants.go:20]: I would prefer to keep the values used in `defaultConfig` to the constant values that are defined above. Perhaps `defaultKeyBindings` could be added as a new const instead?

**Comment by savedra1** [comment]
> I forgot to change back to the defaults after I did some testing, sorry for that silly mistake 😋
> 
> About the fuzzySelect, I didn't look into it, but it was in the original code.

No worries! 

I think `fuzzySelect` is my bad.. not actually being used by anything. It can be dropped entirely from the `keys.go` file and from the new struct :smile:

**Comment by savedra1** [comment]
Another thought @TiagoRibeiro25, do you think it would also be a good idea to apply the custom keybinds to the secondary view screens as well? E.g the confirmation screen/preview screen. All could be handled from the same `keys.go` file.

**Comment by TiagoRibeiro25** [comment]
@savedra1 I overlooked the secondary views initially, but I've now added support for them. I had to adjust some keybindings to better suit the new setup. Another solution would be adding more specific keybindings to the config file just for those screens.

**Review by savedra1 — APPROVED** [review]

**Comment by savedra1** [comment]
Closes #135

**Pull request merged**